src/connection.c

```diff
@@ -325,26 +326,47 @@ static int backoff_delay (backoff_timer_t *timer)
   struct timespec ts;
   int rtn;
 
+  pthread_condattr_t backoff_delay_cond_attr;
+
+  pthread_condattr_init (&backoff_delay_cond_attr);
+  pthread_condattr_setclock (&backoff_delay_cond_attr, CLOCK_MONOTONIC);
+  pthread_cond_init (&backoff_delay_con, &backoff_delay_cond_attr);
+
   // periodically update the health file.
-  clock_gettime (CLOCK_REALTIME, &ts);
+  clock_gettime (CLOCK_MONOTONIC, &ts);
   if ((ts.tv_sec - timer->ts.tv_sec) >= UPDATE_HEALTH_FILE_INTERVAL_SECS) {
     start_conn_in_progress (timer->start_time);
     timer->ts.tv_sec += UPDATE_HEALTH_FILE_INTERVAL_SECS;
   }	  
 
   calc_random_expiration (random(), random(), timer, &ts);
+  ParodusInfo("After calc_random_expiration.\n");
+
+  ParodusInfo("Timer value - start_time : %lu, backoffRetrytime sec : %ld, nsec : %ld, count : %d, max_count : %d, delay - %d\n", timer->start_time, timer->ts.tv_sec, timer->ts.tv_nsec, timer->count, timer->max_count, timer->delay);
```

Comment:
This was only for debugging, we can make it debug level

Reply:
Removed in latest commit. Please have a look.